File: client/src/app/model/Inventory/StoreReceivalListDto.ts
export class StoreReceivalListDto {
    id: string;
    itemName: string;
    requestNumber: string;
    approvedQuantity: number;
    measurementUnitName: string;
    requesterEmployee: string;
    approverEmployee: string;
}

export class StoreRecivalListDto {
    productId: string;
    measurementId: string;
    quantity: string;
    price: string;
}

export class StoreRequestIssueDto {
    id: string;
    userId: string;
    date: Date;
    quantity: number;
}


export class ApprovedItemsDto {
    id: string;
    itemName: string;
    approvedQuantity: number;
    measurementUnit: string;
    aproverEmployee: string;
}

export class ReciveTransportableItems {
    employeeId: string;
    recivalItemId: string[];
}

export class ReceiveItems {
    itemRecivalId: string;
    employeeId: string;
    userId: string;
    rowName: string;
    columnName: string;
}

export class EmployeeReceivedITemsDto {
    id: string;
    itemName: string;
    issuedQuantity: number;
    remainingQuantity: number;
    measurementUnit: string;
    employeeRecivedProducts : EmployeeRecivedProductsDto[];
}

export class EmployeeRecivedProductsDto
{
    id: string;
    productDetailName: string;
    tagNumber: string;
    serialNumber: string;
}


export class AdjustReceivedITemsDto
{
    id: string;
    createdById: string;
    usedQuantity: number;
    remark: string;
    usedItemStatus: number;
}

File: IntegratedDigitalAPI/IntegratedImplementation/DTOS/Inventory/StoreReceivalListDto.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Drawing.Printing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using static IntegratedInfrustructure.Data.EnumList;

namespace IntegratedImplementation.DTOS.Inventory
{
    public class StoreReceivalListDto
    {
        public string Id { get; set; } = null!;
        public string ItemName { get; set; } = null!;
        public string RequestNumber { get; set; } = null!;
        public double ApprovedQuantity { get; set; }
        public string MeasurementUnitName { get; set; } = null!;
        public string RequesterEmployee { get; set; } = null!;
        public string ApproverEmployee { get; set; } = null!;
    }

    public class StoreRecivalListDto
    {
        public Guid ProductId { get; set; }
        public Guid MeasurementId { get; set; }
        public double Quantity { get; set; }
        public double Price { get; set; }
    }

    public class StoreRequestIssueDto
    {
        public string Id { get; set; } = null!;
        public string UserId { get; set; } = null!;
        public int Quantity { get; set; }
    }


    public class ApprovedItemsDto
    {
        public string Id { get; set; } = null!;
        public string ItemName { get; set; } = null!;
        public double ApprovedQuantity { get; set; }
        public string MeasurementUnit { get; set; } = null!;
        public string ApproverEmployee { get; set; } = null!;
    }

    public class ReciveTransportableItems
    {
        public string EmployeeId { get; set; } = null!;
        public List<string> RecivalItemId { get; set; } = null!;
    }

    public class ReceiveItems
    {
        public string ItemRecivalId { get; set; } = null!;
        public string EmployeeId { get; set; } = null!;
        public string UserId { get; set; } = null!;
        public string? RowName { get; set; } = null!;
        public string? ColumnName { get; set; } = null!;
    }

    public class EmployeeReceivedITemsDto
    {
        public string Id { get; set; } = null!;
        public string ItemName { get; set; } = null!;
        public double IssuedQuantity { get; set; }
        public double RemainingQuantity { get; set; }
        public string MeasurementUnit { get; set; } = null!;

        public List<EmployeeRecivedProductsDto> EmployeeRecivedProducts { get; set; } = null!;

    }

    public class EmployeeRecivedProductsDto
    {
        public string Id { get; set; } = null!;
        public string ProductDetailName { get; set; } = null!;
        public string TagNumber { get; set; } = null!;
        public string? SerialNumber { get; set; } = null!;
        public string Status { get; set; } = null!;
    }



    public class AdjustReceivedITemsDto
    {
        [Required]
        public string Id { get; set; } = null!;
        [Required]
        public string CreatedById { get; set; } = null!;
        public string Remark { get; set; } = null!;
        [Required]
        public UsedItemsStatus UsedItemStatus { get; set; }
    }




}
